### PR TITLE
[Tag] Add icon and rightIcon props

### PR DESCRIPTION
**Pull request by llorca**
image

### CODE REVIEW

**Comment by llorca** [comment]
you're right, but since 2.0, you can put any SVG in `icon` props, not just a Blueprint one. I highly discourage using a default 16px icon in there, or resizing a 16px one to 12. On our end, we use custom made 12px icons to ensure crispness

**Comment by llorca** [comment]
for some reason, when showing the left icon only (not the right one), the stray div container's height increases by 2px... any idea how to go around that?

**Comment by giladgray** [comment]
soooo this is already technically possible since you can just put JSX in `<Tag>` elements, so you can do `<Tag><Icon />content<Icon /></Tag>`. all we really need to do is fix the spacing, and i think the best way to do that is along the lines of how `pt-flex-container()` mixin will automatically put margin after all elements except the last.

i'm happy to pick this up at some point next week.

**Comment by giladgray** [comment]
🤔 multiline support is poor here

File: packages/core/src/common/classes.ts [hunk 1]
```diff
@@ -205,7 +205,6 @@ export const TAB_PANEL = `${TAB}-panel`;
 export const TABS = `${TAB}s`;
 
 export const TAG = `${NS}-tag`;
-export const TAG_REMOVABLE = `${TAG}-removable`;
```
Comment (themadcreator) [inline packages/core/src/common/classes.ts]: another breaking change to make note of in release notes

Comment (giladgray) [reply packages/core/src/common/classes.ts]: def, tho it only affects CSS consumers. #2587

**Pull request merged**